### PR TITLE
fix decode for some .nif, and allow external texture to be loaded on embedded .nifs

### PR DESCRIPTION
**Pull request by M4rkFlor**
Added safe_decode function to math utils and changed decode function only where it caused crash. Decode error can still occur where safe decode is not used. Also added some support for embedded .nif textures to be loaded externally

@niftools/blender-niftools-addon-reviewer 

# Overview
**[Fixes a decode bug]**

##  Detailed Description
**[addded safe_decode to utils.math, replaced the decode functions causing crash with safe_decode. The actual encoding of the file is unknown, I tested a list of potential encodings and choose the most accurate, but it is still not correct. Fixed definition error of IMPORT_EMBEDDED_TEXTURES]**

## Fixes Known Issues
**[#441, #421 partially, #443 partially]**

## Documentation
**[None]**

## Testing
**[was able to load models without Decode error, but results in new errors potentially unrelated,  was able to load embedded .nifs with external textures]**

### Manual
**[Imported a .nif and decode error wasn't generated]**

### Automated
**[None]**

## Additional Information
**[The next error is due to IMPORT_EMBEDDED_TEXTURES not being implemented.(fixed). This PR was edited to add another PR]**

### CODE REVIEW

**Review by neomonkeus — COMMENTED** [review]
Great work, just some minor changes needed.

**Comment by M4rkFlor** [comment]
I followed the suggestions. I removed IMPORT_EMBEDDED_TEXTURES and made it a UI option. I Also took niftools/pyffi#68 and integrated it. Now embedded textures work to some extent.

File: io_scene_niftools/modules/nif_import/property/texture/loader.py [hunk 1]
```diff
@@ -42,13 +42,136 @@
...
 import bpy
+from pyffi.formats.dds import DdsFormat
 from pyffi.formats.nif import NifFormat
 
 from io_scene_niftools.modules.nif_import.property import texture
 from io_scene_niftools.utils.singleton import NifOp
 from io_scene_niftools.utils.logging import NifLog
 
 
+# TODO once the save_dds PR code is merged into pyffi this section can be deleted.
```
Comment (neomonkeus) [inline io_scene_niftools/modules/nif_import/property/texture/loader.py:53]: 💯

File: io_scene_niftools/modules/nif_import/property/texture/loader.py [hunk 2]
```diff
@@ -73,8 +196,8 @@ def import_texture_source(self, source):
         # if the source block is not linked then return None
         if not source:
             return None
-        NifLog.warn(f"Import embedded texture skipped, defaulting to import external texture")#IMPORT_EMBEDDED_TEXTURES is currently a constant
-        if isinstance(source, NifFormat.NiSourceTexture) and not source.use_external and texture.IMPORT_EMBEDDED_TEXTURES:
+
+        if isinstance(source, NifFormat.NiSourceTexture) and not source.use_external and NifOp.props.use_embedded_texture:
```
Comment (neomonkeus) [inline io_scene_niftools/modules/nif_import/property/texture/loader.py:200]: 👍

File: io_scene_niftools/ui/operators/nif_import.py [hunk 3]
```diff
@@ -94,7 +94,7 @@ def draw(self, context):
 
 
 class OperatorImportGeometryPanel(OperatorSetting, Panel):
-    bl_label = "Transform"
+    bl_label = "Geometry"
```
Comment (neomonkeus) [inline io_scene_niftools/ui/operators/nif_import.py:97]: Good catch

**Review by neomonkeus — APPROVED** [review]

**Comment by neomonkeus** [comment]
Looks like you went further than my comments. 
Really well polished PR, ready to merge, just holding off as I want to do a release of v0.0.7 without this first so people can test on develop.

**Review by HENDRIX-ZT2 — APPROVED** [review]
Looks great!

**Review by neomonkeus — APPROVED** [review]

File: io_scene_niftools/utils/math.py [hunk 4]
```diff
@@ -227,3 +227,9 @@ def mathutils_to_nifformat_matrix(b_matrix):
     n_matrix = NifFormat.Matrix44()
     n_matrix.set_rows(*b_matrix.transposed())
     return n_matrix
+
+def safe_decode(b: bytes) -> str:
```
[the remark this answers is not in the document]
Comment (neomonkeus) [reply io_scene_niftools/utils/math.py]: Done

File: io_scene_niftools/modules/nif_import/property/texture/__init__.py [hunk 5]
```diff
@@ -36,3 +36,7 @@
 # POSSIBILITY OF SUCH DAMAGE.
 #
 # ***** END LICENSE BLOCK *****
+
```
[the remark this answers is not in the document]
Comment (neomonkeus) [reply io_scene_niftools/modules/nif_import/property/texture/__init__.py]: Done

**Comment by Shasties** [comment]
Found this when looking into issue #421. Needed to add the safe_decode import to io_scene_niftools\modules\nif_import\object\__init__.py and change n_name = safe_decode(n_block.name) on line 70 to get around exceptions. May want to consider adding into this PR.

**Pull request merged**